joplin/api/schema.py

```diff
@@ -340,22 +341,115 @@ class Meta:
         interfaces = [graphene.Node]
 
 
-class ProcessPageStepNode(DjangoObjectType):
-    class Meta:
-        model = ProcessPageStep
-        interfaces = [graphene.Node]
-
-
-class ProcessPageContactNode(DjangoObjectType):
-    class Meta:
-        model = ProcessPageContact
-        interfaces = [graphene.Node]
-
-
-class ProcessPageTopicNode(DjangoObjectType):
-    class Meta:
-        model = ProcessPageTopic
-        interfaces = [graphene.Node]
+class SiteStructure(graphene.ObjectType):
+    value = GenericScalar()
+    structure_json = JSONString()
+
+    # json isn't a great way to do this, we should
+    # figure out how to make it queryable
```

Comment:
this is definitely less than ideal, but it sets in motion a better way of handling things.

for now we're just getting every possible url for any given page id, and returning that with a bit of extra info for contextual nav.